unified-runtime/source/adapters/cuda/usm.cpp

```diff
@@ -569,7 +569,9 @@ urUSMPoolTrimToExp(ur_context_handle_t hContext, ur_device_handle_t hDevice,
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL urUSMContextMemcpyExp(ur_context_handle_t,
-                                                          void *, const void *,
-                                                          size_t) {
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+                                                          void *pDst,
+                                                          const void *pSrc,
+                                                          size_t Size) {
+  UR_CHECK_ERROR(cuMemcpyHtoD((CUdeviceptr)pDst, pSrc, Size));
```

Comment:
Just curious - where in the API does it say that `pSrc` is a host pointer and `pDst` is a device pointer? I can't find much information about the extension.

Reply:
Seems you're right, the docs for the function declaration don't mention it explicitly. Are you worried that this call could potentially handle situation where both src and dest are device pointers?

Reply:
Yeah, or host->host or device->host. It's not clear to me what this API is meant to handle.

Reply:
I've discussed it offline, you are right, the extension does not give any guarantees about the location of the pointers. I'll put it back to draft to address this issue.

Reply:
We should be bale to trace the pointers with [`cuPointerGetAttribute`](https://developer.download.nvidia.cn/compute/DevZone/docs/html/C/doc/html/group__CUDA__UNIFIED_g0c28ed0aff848042bc0533110e45820c.html) and [`CU_POINTER_ATTRIBUTE_MEMORY_TYPE`](https://developer.download.nvidia.cn/compute/DevZone/docs/html/C/doc/html/group__CUDA__TYPES_gb3c95cb65dc7cb3e03e562d129cbb573.html#ggb3c95cb65dc7cb3e03e562d129cbb5730409e16293b60b383f30a9b417b2917c).

Reply:
I've moved to [`cuMemcpy`](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g8d0ff510f26d4b87bd3a51e731e7f698) that automatically handles the provenience of the src/dst pointers.